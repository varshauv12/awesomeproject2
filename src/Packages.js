import React from "react";
import { View, Text ,TouchableOpacity, Image} from 'react-native';
import { Globalstyle } from "./Globalstyle";
import { FontAwesome } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';

const Packages = () => {
    const navigation = useNavigation()  
    return (
            
        <View style={Globalstyle.container}>
        <View style={Globalstyle.headerContainer}>

        <TouchableOpacity style={Globalstyle.iconBackContainer} onPress={() => navigation.goBack()}>
          <FontAwesome name="arrow-left" size={25} color={"white"} />
        </TouchableOpacity>
        <TouchableOpacity style={Globalstyle.iconContact}>
          <FontAwesome name="info-circle" size={15} color={"#EDF6FF"} /><Text style={Globalstyle.iconInfo}>What to send</Text>
          </TouchableOpacity>
        </View>
            
            
        
            <View style={Globalstyle.cntalin} >
                <View>
                <Image
                source={require('../assets/Images/img.png')}
                style={Globalstyle.imgdata} 
              />
                </View>
                <View style={{top:50}}>
                
            <Text style={Globalstyle.primrytext}>Send packages with Connect</Text>
            <Text style={Globalstyle.Packagestxt}>Get it delivered in the time it takes to drive there</Text>

            <TouchableOpacity style={Globalstyle.Packagebtn} onPress={() => {
            navigation.navigate('SendPackages')
            }}>
                <Text style={Globalstyle.Packagebtntxt} >Send a package</Text>
            </TouchableOpacity>
            
            <TouchableOpacity style={Globalstyle.Packagesrvcbtn} onPress={() => {
                navigation.navigate('RecievingPackage')
            }}>
                <Text style={Globalstyle.Packagebtntxt} >Receive a package</Text>
                    </TouchableOpacity>
                    </View>
            </View>
            </View>
    );
};

export default Packages;

// const styles = StyleSheet.create({
    

  
// });
