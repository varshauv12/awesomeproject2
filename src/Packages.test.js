import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import Packages from './Packages';
// Mock @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
    FontAwesome: 'FontAwesome', // Mock the specific icon component you're using
}));

// Mock useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('AccountHome', () => {
  it('renders correctly', () => {
    // Mock useNavigation behavior
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    // Render the Login component
    const tree = renderer.create(<Packages/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});