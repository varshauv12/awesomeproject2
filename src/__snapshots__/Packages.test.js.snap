// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountHome renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1A1A1A",
      "flex": 1,
      "textAlign": "center",
      "top": 0,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-start",
        "opacity": 1,
        "right": 150,
        "top": 15,
      }
    }
  >
    <FontAwesome
      color="white"
      name="arrow-left"
      size={25}
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "color": "#EDF6FF",
        "left": 90,
        "opacity": 1,
      }
    }
  >
    <FontAwesome
      color="#EDF6FF"
      name="info-circle"
      size={15}
    />
    <Text
      style={
        {
          "bottom": 17,
          "color": "#EDF6FF",
          "left": 20,
        }
      }
    >
      What to send
    </Text>
  </View>
  <View>
    <View>
      <Image
        source={
          {
            "testUri": "../../../assets/Images/img.png",
          }
        }
        style={
          {
            "height": 200,
            "left": 0,
            "top": 40,
            "width": 300,
          }
        }
      />
    </View>
    <View
      style={
        {
          "top": 50,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 30,
            "top": 55,
          }
        }
      >
        Send packages with Connect
      </Text>
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 16,
            "top": 85,
            "width": 295,
          }
        }
      >
        Get it delivered in the time it takes to drive there
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderWidth": 1,
            "color": "#FFFFFF",
            "height": 50,
            "opacity": 1,
            "top": 130,
            "width": 310,
          }
        }
      >
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 20,
              "paddingTop": 8,
            }
          }
        >
          Send a package
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderWidth": 1,
            "color": "#FFFFFF",
            "height": 50,
            "opacity": 1,
            "top": 145,
            "width": 310,
          }
        }
      >
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 20,
              "paddingTop": 8,
            }
          }
        >
          Receive a package
        </Text>
      </View>
    </View>
  </View>
</View>
`;
